test/tests/classTransformer.test.ts

```diff
@@ -0,0 +1,118 @@
+import { ObjectId } from 'bson';
+import { classToPlain, Exclude, Expose, plainToClass, Transform } from 'class-transformer';
+
+import { prop } from '../../src/prop';
+import { getModelForClass } from '../../src/typegoose';
+
+// re-implement base Document to allow class-transformer to serialize/deserialize its properties
+class Document {
```

Comment:
would it be possible to change the name of this class to something like `DocumentCT` or would this defeat the purpose of "overwriting" mongoose-Document?

Reply:
I just used matching name for no ambiguity, but yes let me rename it.